packages/ve-hemi-actions/actions/public/veHemi.ts

```diff
@@ -61,3 +62,45 @@ export const getOwnerOf = async function (
     functionName: 'ownerOf',
   })
 }
+
+export const getBalanceOfNFTAt = async function (
+  client: Client,
+  tokenId: bigint,
+  timestamp: bigint,
+): Promise<bigint> {
+  if (!client.chain) {
+    throw new Error('Client chain is not defined')
+  }
+
+  const veHemiAddress = getVeHemiContractAddress(client.chain.id)
+
+  return readContract(client, {
+    abi: veHemiAbi,
+    address: veHemiAddress,
+    args: [tokenId, timestamp],
+    functionName: 'balanceOfNFTAt',
+  })
+}
+
+export const getTotalVeHemiSupplyAt = async function (
+  client: Client,
+  timestamp: bigint,
+): Promise<bigint> {
+  if (!client.chain) {
+    throw new Error('Client chain is not defined')
+  }
+
+  const veHemiAddress = getVeHemiContractAddress(client.chain.id)
+
+  // Determine function name based on chain ID
+  // because the function name differs between Hemi mainnet and Hemi Sepolia contracts
+  const functionName =
+    client.chain.id === hemi.id ? 'totalVeHemiSupplyAt' : 'totalSupplyAt'
```

Comment:
What version of the Hemi Sepolia contract are we using? [MockVeHemi](https://testnet.explorer.hemi.xyz/address/0xd137b7B3510b23E98b6A9aC2acB5C8bb668AC82d) has `totalVeHemiSupplyAt` implemented.